Mathlib/LinearAlgebra/QuadraticForm/Basis.lean

```diff
@@ -16,9 +18,125 @@ a basis.
 
 open LinearMap (BilinMap)
 
+section
+
+variable {ι R M} [Semiring R] [AddCommMonoid M] [Module R M]
+
+/--
+Define the multiplication of a scalar-valued, finitely supported, function with a module-valued
+function as a finitely supported module-valued function.
+-/
+noncomputable def Finsupp.pointwiseSMul (f : ι →₀ R) (g : ι → M) : ι →₀ M :=
+    Finsupp.ofSupportFinite (fun a ↦ f a • g a) (by
```

Comment:
Everywhere you use this new definition, you use it as a function, never as a finsupp. Therefore I doubt you need it, and instead you can use `f • g` everywhere

Reply:
Presumably `((Sym2.mulFinsupp l) : Sym2 ι → R)` should be `Sym2.mul ∘ Sym2.map l` as well.

Reply:
Yes, very likely

Reply:
Okay, I think I've fixed that, thanks.